hack/README.md

```diff
@@ -0,0 +1,138 @@
+# Kubernetes hack GuideLines
+
+This document describes how you can use the scripts from [`hack`](.) directory 
+and gives a brief introduction and explanation of these scripts. 
+
+## Overview
+
+The [`hack`](.) directory contains many scripts that ensure continuous development of kubernetes, 
+enhance the robustness of the code, improve development efficiency, etc. 
+The explanations and descriptions of these scripts are helpful for contributors. 
+For details, refer to the following guidelines.
+
+## Key scripts
```

Comment:
This seems to just list every file, can you pick "key" scripts?

How about if we just tell people to run `verify-all.sh` before submitting a PR and if anything fails run `update-all.sh`?

Anyone interested in more detail should probably just look at the scripts themselves.

Listing every script this way is going to be out of date almost as soon as its submitted, and manually making a list of every file doesn't do anything particularly different from what github already does.

Reply:
I think it's a good idea to pick up only key scripts and add guidance for each category(verify, update and etc.) in readme.md. 
Are `verify-all.sh`  and `update-all.sh` the only key scripts?

Reply:
Yes, as they run all the verify and update scripts respectively.

Reply:
I still think this could separate out update-all and verify-all into their own primary section

Reply:
sure i will update it